src/gpu/optixPrograms.cu

```diff
@@ -163,21 +166,49 @@ extern "C" __global__ void __closesthit__()
 	}
 
 	float intensity = 0;
-	if (sbtData.textureCoords != nullptr && sbtData.texture != 0) {
-		assert(index.x() < sbtData.textureCoordsCount);
-		assert(index.y() < sbtData.textureCoordsCount);
-		assert(index.z() < sbtData.textureCoordsCount);
+	if (entityData.textureCoords != nullptr && entityData.texture != 0) {
+		assert(triangleIndices.x() < entityData.textureCoordsCount);
+		assert(triangleIndices.y() < entityData.textureCoordsCount);
+		assert(triangleIndices.z() < entityData.textureCoordsCount);
 
-		const Vec2f& uvA = sbtData.textureCoords[index.x()];
-		const Vec2f& uvB = sbtData.textureCoords[index.y()];
-		const Vec2f& uvC = sbtData.textureCoords[index.z()];
+		const Vec2f& uvA = entityData.textureCoords[triangleIndices.x()];
+		const Vec2f& uvB = entityData.textureCoords[triangleIndices.y()];
+		const Vec2f& uvC = entityData.textureCoords[triangleIndices.z()];
 
 		Vec2f uv = (1 - u - v) * uvA + u * uvB + v * uvC;
 
-		intensity = tex2D<TextureTexelFormat>(sbtData.texture, uv[0], uv[1]);
+		intensity = tex2D<TextureTexelFormat>(entityData.texture, uv[0], uv[1]);
+	}
+
+	Vec3f velocity{NAN};
+	Vec3f transformDisplacement = {0, 0, 0};
+	Vec3f skinningDisplacement = {0, 0, 0};
+	if (ctx.sceneDeltaTime > 0) {
+		if (entityData.hasPrevFrameLocalToWorld) {
+			// Computing hit point velocity in simple words:
+			// From raytracing, we get hit point in Entity's coordinate frame (hitObject).
+			// Think of it as a marker dot on the Entity.
+			// Having access to Entity's previous pose, we can compute (entityData.prevFrameLocalToWorld * hitObject),
+			// where the marker dot would be in the previous raytracing frame (displacementOriginWorld).
+			// Then, we can connect marker dot in previous raytracing frame with its current position and obtain transformDisplacement vector
+			// Dividing transformDisplacement by time elapsed from the previous raytracing frame yields velocity vector.
+			Vec3f displacementOriginWorld = entityData.prevFrameLocalToWorld * hitObject;
```

Comment:
`displacementOriginWorld ` is very misleading.
This is just globalHitpointPosition. Not displacement

Reply:
or worldHitpointPosition if U want.

Reply:
`DisplacementVectorOrigin`